compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -642,7 +643,7 @@ fn codegen_regular_intrinsic_call<'tcx>(
             if intrinsic == sym::assert_zero_valid
                 && !fx
                     .tcx
-                    .permits_zero_init(fx.param_env().and(ty))
+                    .check_validity_of_init(fx.param_env().and((ty::InitKind::Zero, ty)))
```

Comment:
Unless you expect `InitKind` to have type info in it, this probably is better as `(ty::InitKind::Zero, param_env.and(ty))` as a key (just easier to understand imo)